### PR TITLE
Add PriorityClass for operator pod

### PR DESCRIPTION
**Pull request by elsnepal**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
Helm chart now supports specifying priorityClassName in values.yaml
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by caseydavenport** [comment]
@elsnepal thanks for the PR!

I think you will need to run `make generate` to update files that are generated from the helm chart in order to make CI pass.

**Comment by elsnepal** [comment]
> @elsnepal thanks for the PR!
> 
> I think you will need to run `make generate` to update files that are generated from the helm chart in order to make CI pass.

@caseydavenport is there any docs around required version for me to be able to run that?
➜  calico git:(master) ✗ make --version
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
➜  calico git:(master) ✗ make generate
/Library/Developer/CommandLineTools/usr/bin/make gen-semaphore-yaml
cd .semaphore && ./generate-semaphore-yaml.sh
sed: 1: "semaphore.yml": unterminated substitute in regular expression
sed: 1: "semaphore-scheduled-bui ...": bad flag in substitute command: 'd'
make[1]: *** [gen-semaphore-yaml] Error 1
make: *** [generate] Error 2

**Comment by caseydavenport** [comment]
@elsnepal ah, looks like you're running on a mac and the `sed` command might be functioning differently. 

This is all that script is doing: https://github.com/projectcalico/calico/blob/master/.semaphore/generate-semaphore-yaml.sh

You might need to install GNU sed for mac

**Comment by elsnepal** [comment]
@caseydavenport Sorry i forgot to get back, i did make local switch `sed` to `gsed` and stumbled at this:

```
F0124 14:00:34.819366      17 deepcopy.go:890] Hit an unsupported type invalid type for invalid type, from github.com/projectcalico/api/pkg/apis/projectcalico/v3.BGPConfiguration
make[1]: *** [gen-files] Error 255
make: *** [generate] Error 2
```

➜  calico git:(patch-1) ✗ echo $GOROOT
/usr/local/Cellar/go/1.21.3/libexec

**Comment by elsnepal** [comment]
Anything i could do to get this reviewed and merged?

**Comment by caseydavenport** [comment]
@elsnepal Sorry for the delay - let me see if I can figure out what is going on. Just need to get `make generate` working.

File: charts/tigera-operator/templates/tigera-operator/02-tigera-operator.yaml [hunk 1]
```diff
@@ -32,6 +32,7 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       serviceAccountName: tigera-operator
+      priorityClassName: {{.Values.priorityClassName}}
```
Comment (caseydavenport) [inline charts/tigera-operator/templates/tigera-operator/02-tigera-operator.yaml:35]:
```suggestion
{{- if .Values.priorityClassName }}
      priorityClassName: {{.Values.priorityClassName}}
{{- end }}
```

**Comment by caseydavenport** [comment]
It looks like our generation code won't work very well on a mac, so I've taken your changes and added an additional commit on top that gets past that hopefully.

**Review by caseydavenport — APPROVED** [review]

**Pull request merged**